p2p/host/autonat/autonat.go

```diff
@@ -198,15 +199,20 @@ func (as *AmbientAutoNAT) background() {
 			}
 
 		// probe finished.
-		case result, ok := <-as.observations:
+		case err, ok := <-as.dialResponses:
 			if !ok {
 				return
 			}
-			as.recordObservation(result)
+			if IsDialRefused(err) {
+				as.retryProbe = true
```

Comment:
Why not just pass this as a parameter to `scheduleProbe`?

Reply:
I changed it to pass it as a parameter but why is this better? 

```scheduleProbe``` is already making the scheduling decision based on params set on the struct like ```as.lastInbound``` why handle this info separately?

Reply:
Usually I try to keep as little members as possible, but you're right, there seems to be precedent here for not following this principle.

Reply:
I agree with your logic, it is better to have this state have as little visibility as possible. We can keep this and change the other "precedents" later when we touch that piece.